src/frontend/hooks/hasStatus.ts

```diff
@@ -9,7 +9,7 @@ import libraryState from 'frontend/state/libraryState'
 // the consuming code needs to be wrapped in observer when using this hook
 export const hasStatus = (
   appName: string,
-  gameInfo: GameInfo,
+  gameInfo?: GameInfo,
```

Comment:
Why the change here?
Will it work if we do not pass gameinfo here?

Reply:
Initially, it was from @BrettCleary  as the following, therefore just changed. I believe this I changed as soon I went on this PR 

```
export const hasStatus = (
  appName: string,
  gameInfo: GameInfo | undefined,
```

Reply:
@red-game-dev @flavioislima looks like this propagates from src/frontend/component/UI/DownloadToastManager where the currentElement could be undefined. It is set to undefined after a timeout when it finishes installing to reset the toast. When currentElement or gameInfo are undefined, `<></>` is returned so no worries about showing the wrong state on the download toast

iirc we can't return earlier than `hasStatus` otherwise we change the number of hooks rendered on subsequent re-renders when a download starts

Honestly we should just move all this state and logic into a mobx class and add tests for it. Then we wouldn't have to deal with these annoying react limitations. This code was written before we committed to mobx.